### PR TITLE
Bug 1254441 - Fixed section header constraints to allow dynamic header sizing in settings

### PR DESCRIPTION
**Pull request by sleroux**

### CODE REVIEW

File: Client/Frontend/Settings/SettingsTableViewController.swift [hunk 1]
```diff
@@ -554,4 +524,23 @@ class SettingsTableSectionHeaderFooterView: UITableViewHeaderFooterView {
         titleLabel.text = nil
         titleAlignment = .Bottom
     }
+
+    private func remakeTitleAlignmentConstraints() {
+        switch titleAlignment {
+        case .Top:
+            titleLabel.snp_remakeConstraints { make in
+                make.left.equalTo(self).offset(SettingsTableSectionHeaderFooterViewUX.titleHorizontalPadding)
+                make.right.lessThanOrEqualTo(self).offset(-SettingsTableSectionHeaderFooterViewUX.titleHorizontalPadding)
```
Comment (thebnich) [inline Client/Frontend/Settings/SettingsTableViewController.swift]: Why do we need `lessThanOrEqualTo` for `right`?

File: Client/Frontend/Settings/SettingsTableViewController.swift [hunk 2]
```diff
@@ -554,4 +524,23 @@ class SettingsTableSectionHeaderFooterView: UITableViewHeaderFooterView {
...
+                make.top.equalTo(self).offset(SettingsTableSectionHeaderFooterViewUX.titleVerticalPadding)
+                make.bottom.equalTo(self).offset(-SettingsTableSectionHeaderFooterViewUX.titleVerticalLongPadding)
+            }
```
Comment (thebnich) [inline Client/Frontend/Settings/SettingsTableViewController.swift]: SnapKit supports `.offset(UIEdgeInsets)`, and also `.inset(UIEdgeInsets)` (haven't looked at the difference carefully). Can we turn all of these into a one-liner using one of those?

File: Client/Frontend/Settings/SettingsTableViewController.swift [hunk 3]
```diff
@@ -554,4 +524,23 @@ class SettingsTableSectionHeaderFooterView: UITableViewHeaderFooterView {
...
+        case .Bottom:
+            titleLabel.snp_remakeConstraints { make in
+                make.left.equalTo(self).offset(SettingsTableSectionHeaderFooterViewUX.titleHorizontalPadding)
+                make.right.lessThanOrEqualTo(self).offset(-SettingsTableSectionHeaderFooterViewUX.titleHorizontalPadding)
+                make.bottom.equalTo(self).offset(-SettingsTableSectionHeaderFooterViewUX.titleVerticalPadding)
+                make.top.equalTo(self).offset(SettingsTableSectionHeaderFooterViewUX.titleVerticalLongPadding)
+            }
```
Comment (thebnich) [inline Client/Frontend/Settings/SettingsTableViewController.swift]: Same questions.

**Comment by sleroux** [comment]
I was able to combine the two lines into one line using `insets`. Also I thought I had the lessThanOrEqual because of an issue with ambiguous constraints but testing various devices it doesn't seem to be an issue. Maybe not that the header is using autolayout to calculate it's size it's not longer an issue?

**Pull request merged**